.circleci/config.in.yml

```diff
@@ -253,3 +301,19 @@ workflows:
     jobs:
       - circleci_consistency
       {{ workflows_nightly }}
+  test-promote:
+    # These are all dry-run by default
+    jobs:
+      # TODO: find a way to build the dockerfile per-pr
```

Comment:
We talked about priorities for what order to do things, but I think the proposed long docker build time should only take a ~day to prototype out in this context. Worth trying, I think!